src/Illuminate/Validation/Validator.php

```diff
@@ -890,7 +890,7 @@ protected function validateInteger($attribute, $value)
             return true;
         }
 
-        return is_null($value) || filter_var($value, FILTER_VALIDATE_INT) !== false;
+        return is_null($value) || ctype_digit(ltrim((string) $value, '-'));
```

Comment:
`return is_null($value) || preg_match('/^[\+-]?\d+$/s', $value);` would be faster and would also match and `+` prefix